File: bank-account-app-front/src/app/shared/models/backend/account/account-response.dto.ts
import { CustomerResponseDto } from "../customer/customer-response.dto";

export interface AccountResponseDto {
    accountId: string;
    accountType: string;
    accountState: string;
    balance: number;
    overdraft: number;
    interestRate: number;
    createdAt: Date;
    customerResponseDto: CustomerResponseDto;
}

File: bank-account-app-back/exalt-hexagonal-archi-kafka-keycloak-bs-ms-operation-api/src/main/java/com/exalt/exalthexagonalarchikafkakeycloakbsmsoperationapi/infrastructure/adapters/feign_clients/domain/AccountResponseDto.java
package com.exalt.exalthexagonalarchikafkakeycloakbsmsoperationapi.infrastructure.adapters.feign_clients.domain;


import com.fasterxml.jackson.annotation.JsonInclude;

import java.math.BigDecimal;
import java.time.Instant;
@JsonInclude(JsonInclude.Include.NON_EMPTY)
public record AccountResponseDto (String accountId, String accountType, String accountState, BigDecimal balance,
                                  Double overdraft, Double interestRate,Instant createdAt, CustomerResponseDto customerResponseDto){
}